pkg/apis/config/store.go

```diff
@@ -45,17 +45,26 @@ func FromContext(ctx context.Context) *Config {
 // FromContextOrDefaults is like FromContext, but when no Config is attached it
 // returns a Config populated with the defaults for each of the Config fields.
 func FromContextOrDefaults(ctx context.Context) *Config {
-	if cfg := FromContext(ctx); cfg != nil {
-		return cfg
+	cfg := FromContext(ctx)
+	if cfg == nil {
+		cfg = &Config{}
 	}
-	defaults, _ := NewDefaultsConfigFromMap(map[string]string{})
-	features, _ := NewFeaturesConfigFromMap(map[string]string{})
-	autoscaler, _ := autoscalerconfig.NewConfigFromMap(map[string]string{})
-	return &Config{
-		Defaults:   defaults,
-		Features:   features,
-		Autoscaler: autoscaler,
+
+	if cfg.Defaults == nil {
+		defaults, _ := NewDefaultsConfigFromMap(map[string]string{})
+		cfg.Defaults = defaults
+	}
+
+	if cfg.Features == nil {
+		features, _ := NewFeaturesConfigFromMap(map[string]string{})
+		cfg.Features = features
+	}
+
+	if cfg.Autoscaler == nil {
+		autoscaler, _ := autoscalerconfig.NewConfigFromMap(map[string]string{})
+		cfg.Autoscaler = autoscaler
 	}
+	return cfg
```

Comment:
This fixes a case where cfg is initialized in an autoscaler test with a context containing only autoscaling config. As a result Features is never populated and can panic when referenced. I thought it made sense to stop this from ever happening but if the preference is to add defensive code where Features is being used/or fix the autoscaler test I can make changes accordingly.